### ISSUE TITLE
wa-tree-item should allow anchors to cover the full interactive row

### ISSUE THREAD

**Issue by alirezabeygijelva** [shoelace-style/webawesome#1552]
When we render navigation links with <a> inside <wa-tree-item>, the anchor only occupies the text label. Clicking the whitespace (indentation area) still triggers the tree selection but doesn't follow the link, so users can’t middle-click or Ctrl/Cmd-click anywhere in the row to open the target in a new tab.

  - Web Awesome 3.0.0-beta.6 (via CDN)
  - Browser: Chrome 140.0.7339.208
  - OS: Windows 11

 ### Steps to Reproduce
  1. Render a simple tree with anchors:

     ```html
     <wa-tree selection="leaf">
       <wa-tree-item>
         Section
         <wa-tree-item>
           <a href="/dest">Destination</a>
         </wa-tree-item>
       </wa-tree-item>
     </wa-tree>

  2. Click inside the leaf item but outside the anchor text (e.g., indentation area or trailing whitespace).
  3. Try middle-clicking or Ctrl/Cmd-clicking in that whitespace.

 ### Actual Behaviour

  - The tree item highlights/activates, but the anchor isn’t triggered.
  - Modifier clicks only work when the pointer is precisely over the text node inside the anchor.

  ### Expected Behaviour

  - The anchor (or another built-in option) should cover the entire interactive surface of the tree item so any click inside the row behaves like a link.
  - Modifier clicks (Ctrl/Cmd, Shift, middle mouse) should work anywhere in the item, matching standard navigation UX.

  ### Impact

  Difficult to use tree-based navigation menus because users lose the ability to open destinations in new tabs from the navigation pane without pixel-perfect clicks.

**Comment by gregjotau** [comment]
Example of UI where it is not working: 

<img width="639" height="1039" alt="Image" src="https://github.com/user-attachments/assets/77370da9-158b-4970-856e-67ae97e3d170" />
